pyppeteer/jshandle.py

```diff
@@ -0,0 +1,613 @@
+            }''', self, propertyName)
+        properties = await objectHandle.getProperties()
+        result = properties[propertyName]
+        await objectHandle.dispose()
+        return result
+
+    async def getProperties(self) -> Dict[str, 'JSHandle']:
+        """Get all properties of this handle."""
+        response = await self._client.send('Runtime.getProperties', {
+            'objectId': self._remoteObject.get('objectId', ''),
+            'ownProperties': True,
+        })
+        result = dict()
+        for prop in response['result']:
+            if not prop.get('enumerable'):
+                continue
+            result[prop.get('name')] = createJSHandle(self._context, prop.get('value'))
+        return result
+
+    async def jsonValue(self) -> Dict:
+        """Get Jsonized value of this object."""
+        objectId = self._remoteObject.get('objectId')
+        if objectId:
+            response = await self._client.send(
+                'Runtime.callFunctionOn',
+                {
+                    'functionDeclaration': 'function() { return this; }',
+                    'objectId': objectId,
+                    'returnByValue': True,
+                    'awaitPromise': True,
+                })
+            return helper.valueFromRemoteObject(response['result'])
+        return helper.valueFromRemoteObject(self._remoteObject)
+
+    def asElement(self) -> Optional['ElementHandle']:
+        """Return either null or the object handle itself."""
+        return None
+
+    async def dispose(self) -> None:
+        """Stop referencing the handle."""
+        if self._disposed:
+            return
+        self._disposed = True
+        await helper.releaseObject(self._client, self._remoteObject)
+
+    def toString(self) -> str:
+        """Get string representation."""
+        if self._remoteObject.get('objectId'):
+            _type = self._remoteObject.get('subtype') or self._remoteObject.get('type')
+            return f'JSHandle@{_type}'
+        return 'JSHandle:{}'.format(helper.valueFromRemoteObject(self._remoteObject))
+
+
+class ElementHandle(JSHandle):
+
+    def __init__(
+            self,
+            context: 'ExecutionContext',
+            client: 'CDPSession',
+            remoteObject: 'RemoteObject',
+            page: 'Page',
+            frameManager: 'FrameManager',
+    ):
+        super().__init__(context, client, remoteObject)
+        self._page = page
+        self._frameManager = frameManager
+        self._disposed = False
+
+    def asElement(self) -> Optional['ElementHandle']:
+        return self
+
+    async def contentFrame(self):
+        nodeInfo = await self._client.send(
+            'DOM.describeNode',
+            {
+                'objectId': self._remoteObject.get('objectId')
+            }
+        )
+        frameId = nodeInfo.get('node', {}).get('frameId')
+        if not isinstance(frameId, str):
+            return
+        return self._frameManager.frame(frameId)
```

Comment:
I prefer this way, but it's not that big of deal
```suggestion
        if isinstance(frameId, str):
            return self._frameManager.frame(frameId)
```